fence/sync/sync_users.py

```diff
@@ -65,6 +71,150 @@ def arborist_role_for_permission(permission):
+
+    Returns:
+        Generator[file-like class]: file like object for the file
+    """
+
+    if encrypted:
+        has_crypt = sp.call(["which", "crypt"])
+        if has_crypt != 0:
+            if logger:
+                logger.error("Need to install crypt to decrypt files from dbgap")
+            exit(1)
+        p = sp.Popen(
+            ["crypt", key],
+            stdin=open(filepath, "r"),
+            stdout=sp.PIPE,
+            stderr=open(os.devnull, "w"),
+        )
+        yield StringIO(p.communicate()[0])
+    else:
+        f = open(filepath, "r")
+        yield f
+        f.close()
+
+
+class UserYAML(object):
+    """
+    Representation of the information in a YAML file describing user, project, and RBAC
+    information for access control.
+    """
+
+    def __init__(
+        self, projects=None, user_info=None, policies=None, rbac=None, logger=None,
+        user_rbac=None,
+    ):
+        self.projects = projects or {}
+        self.user_info = user_info or {}
+        self.user_rbac = user_rbac or {}
+        self.policies = policies or {}
+        self.rbac = rbac or {}
+        self.logger = logger
+
+
+    @classmethod
+    def from_file(cls, filepath, encrypted=True, key=None, logger=None):
+        with _read_file(filepath, encrypted=encrypted, key=key, logger=logger) as f:
+            data = yaml.safe_load(f)
+        projects = dict()
+        user_info = dict()
+        policies = dict()
+
+        users = data.get("users", {})
+        for username, details in users.iteritems():
+            # users should occur only once each; skip if already processed
+            if username in projects:
+                if logger:
+                    logger.error("user `{}` occurs multiple times".format(username))
+                raise EnvironmentError("invalid yaml file")
+
+            privileges = {}
+            try:
+                for project in details.get("projects", {}):
+                    privileges[project["auth_id"]] = set(project["privilege"])
+            except KeyError as e:
+                if logger:
+                    logger.error("project missing field: {}".format(e))
+                continue
+
+            user_info[username] = {
+                "email": details.get("email", username),
+                "display_name": details.get("display_name", ""),
+                "phone_number": details.get("phone_number", ""),
+                "tags": details.get("tags", {}),
+                "admin": details.get("admin", False),
+            }
+            projects[username] = privileges
+
+            # list of policies we want to grant to this user, which get sent to arborist
+            # to check if they're allowed to do certain things
+            policies[username] = details.get("policies", [])
+
+
+        # resources should be the resource tree to construct in arborist
+        user_rbac = dict()
+        for username, user_detail in users.iteritems():
```

Comment:
```suggestion
        for username, details in users.iteritems():
```

minor consistency change to match above loop